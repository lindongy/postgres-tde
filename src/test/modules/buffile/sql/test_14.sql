BEGIN;
SELECT buffile_create();
SELECT buffile_seek(0, 8191);
-- Write data across block boundary.
SELECT buffile_write('xyz');
SELECT buffile_seek(0, 8191);
-- Check it's there
SELECT buffile_read(4);
SELECT buffile_seek(0, 8191);
-- Overwrite only part of it.
SELECT buffile_write('ab');
-- Check that the remaining part is not affected, i.e. the 2nd block was
-- loaded before we started to overwrite its contents.
SELECT buffile_seek(0, 8191);
SELECT buffile_read(4);
SELECT buffile_close();
COMMIT;
