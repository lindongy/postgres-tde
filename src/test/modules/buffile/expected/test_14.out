BEGIN;
SELECT buffile_create();
 buffile_create 
----------------
 
(1 row)

SELECT buffile_seek(0, 8191);
 buffile_seek 
--------------
            0
(1 row)

-- Write data across block boundary.
SELECT buffile_write('xyz');
 buffile_write 
---------------
             3
(1 row)

SELECT buffile_seek(0, 8191);
 buffile_seek 
--------------
            0
(1 row)

-- Check it's there
SELECT buffile_read(4);
 buffile_read 
--------------
 \x78797a
(1 row)

SELECT buffile_seek(0, 8191);
 buffile_seek 
--------------
            0
(1 row)

-- Overwrite only part of it.
SELECT buffile_write('ab');
 buffile_write 
---------------
             2
(1 row)

-- Check that the remaining part is not affected, i.e. the 2nd block was
-- loaded before we started to overwrite its contents.
SELECT buffile_seek(0, 8191);
 buffile_seek 
--------------
            0
(1 row)

SELECT buffile_read(4);
 buffile_read 
--------------
 \x61627a
(1 row)

SELECT buffile_close();
 buffile_close 
---------------
 
(1 row)

COMMIT;
